luigi/contrib/beam_dataflow.py

```diff
@@ -0,0 +1,454 @@
+        cmd_line = self.dataflow_executable()
+
+        cmd_line.extend(self._get_dataflow_args())
+        cmd_line.extend(self.args())
+        cmd_line.extend(self._format_input_args())
+        cmd_line.extend(self._format_output_args())
+        return cmd_line
+
+    def _get_runner(self):
+        if not self.runner:
+            logger.warning("Runner not supplied to BeamDataflowJobTask. " +
+                           "Defaulting to DirectRunner.")
+            return "DirectRunner"
+        elif self.runner in [
+            "DataflowRunner",
+            "DirectRunner"
+        ]:
+            return self.runner
+        else:
+            raise ValueError("Runner %s is unsupported." % self.runner)
+
+    def _get_dataflow_args(self):
+        def f(key, value):
+            return '--{}={}'.format(key, value)
+
+        output = []
+
+        output.append(f(self.dataflow_params.runner, self._get_runner()))
+
+        if self.project:
+            output.append(f(self.dataflow_params.project, self.project))
+        if self.zone:
+            output.append(f(self.dataflow_params.zone, self.zone))
+        if self.region:
+            output.append(f(self.dataflow_params.region, self.region))
+        if self.staging_location:
+            output.append(f(self.dataflow_params.staging_location, self.staging_location))
+        if self.temp_location:
+            output.append(f(self.dataflow_params.temp_location, self.temp_location))
+        if self.gcp_temp_location:
+            output.append(f(self.dataflow_params.gcp_temp_location, self.gcp_temp_location))
+        if self.num_workers:
+            output.append(f(self.dataflow_params.num_workers, self.num_workers))
+        if self.autoscaling_algorithm:
+            output.append(f(self.dataflow_params.autoscaling_algorithm, self.autoscaling_algorithm))
+        if self.max_num_workers:
+            output.append(f(self.dataflow_params.max_num_workers, self.max_num_workers))
+        if self.disk_size_gb:
+            output.append(f(self.dataflow_params.disk_size_gb, self.disk_size_gb))
+        if self.worker_machine_type:
+            output.append(f(self.dataflow_params.worker_machine_type, self.worker_machine_type))
+        if self.worker_disk_type:
+            output.append(f(self.dataflow_params.worker_disk_type, self.worker_disk_type))
+        if self.network:
+            output.append(f(self.dataflow_params.network, self.network))
+        if self.subnetwork:
+            output.append(f(self.dataflow_params.subnetwork, self.subnetwork))
+        if self.job_name:
+            output.append(f(self.dataflow_params.job_name, self.job_name))
+        if self.service_account:
+            output.append(f(self.dataflow_params.service_account, self.service_account))
+        if self.labels:
+            output.append(f(self.dataflow_params.labels, json.dumps(self.labels)))
+
+        return output
+
+    def _format_input_args(self):
+        job_input = self.input()
```

Comment:
I think we can apply https://github.com/spotify/luigi/blob/master/luigi/task.py#L856 on `job_input` to get a flattened list so that we don't need to handle it ourselves.

Reply:
Scratch that. I guess it doesn't fit our use case here.